packages/core/src/hooks/useSelect/index.ts

```diff
@@ -301,9 +327,10 @@ export const useSelect = <
       setSearch([]);
       return;
     }
+
     setSearch([
       {
-        field: optionLabel,
+        field: optionLabel as string,
```

Comment:
I think we can add a new prop named `searchLabel` and make `optionLabel` the fallback value if its a string. Default value will be `"title"` (like `optionLabel`) but if a `searchLabel` is passed then it will use it (should be `key TData`).

something like this:

```suggestion
        field: searchLabel ?? (typeof optionLabel === "string" ? optionLabel : "title")
```

and ensure that we're prioritizing `searchLabel` and then `optionLabel` with type string  and if not, fallback to default `optionLabel` which is `"title"`. 

But before making any changes, let's validate this thought 😅 @alicanerdurmaz @BatuhanW

While we're adding `searchLabel`, I think it's best to also add `searchOperator` which is now hardcoded to `"contains"`. When using a custom `optionLabel` with function type, these don't really make sense to me to use.